pkg/apis/pipeline/v1beta1/taskrun_validation.go

```diff
@@ -93,6 +95,43 @@ func (ts *TaskRunSpec) Validate(ctx context.Context) (errs *apis.FieldError) {
 	return errs
 }
 
+// validateInlineParameters validates parameters that are defined inline.
+// This is cruicial for propagated parameters since the parameters could
+// be defined under taskRun and then called directly in the task steps.
+// In this case, paratemers cannot be validated by the underlying taskSpec
+// since they may not have the parameters declared because of propagation.
+func (ts *TaskRunSpec) validateInlineParameters(ctx context.Context) (errs *apis.FieldError) {
+	errs = errs.Also(ValidateParameters(ctx, ts.Params).ViaField("params"))
+	params := ts.Params
+	var paramSpec []ParamSpec
+	if ts.Params == nil || ts.TaskSpec == nil {
+		return errs
+	}
+	for _, p := range params {
+		pSpec := ParamSpec{
+			Name:    p.Name,
+			Default: &p.Value,
+		}
+		paramSpec = append(paramSpec, pSpec)
+	}
+	for _, p := range ts.TaskSpec.Params {
+		skip := false
```

Comment:
would a set be useful here? we use `sets.string` within the codebase. You could also do a map of param name to default.

Reply:
I don't think this comment was addressed but please correct me if I'm wrong

Reply:
The idea behind this comment is that you are sort of implementing a set intersection on param names. However I think what you have is fine, not sure if there's a cleaner way.